src/org/labkey/targetedms/parser/SkylineBinaryParser.java

```diff
@@ -254,26 +253,43 @@ public int matchTransitions(ChromGroupHeaderInfo header, List<? extends GeneralT
                 return match;
         }
 
-        for (GeneralTransition transition : transitions)
+        var start = header.getStartTransitionIndex();
+        var end = start + header.getNumTransitions();
+        var numChromTransitions = end - start + 1;
+        var chromTransitionPosition = _cacheHeaderStruct.getLocationTransitions() + ((long) _cacheHeaderStruct.getChromTransitionSize() * start);
+        ChromTransition[] chromTransitions;
+
+        try
+        {
+            _channel.position(_cacheHeaderStruct.getLocationTransitions() + ((long) _cacheHeaderStruct.getChromTransitionSize() * start));
+             chromTransitions = parseTransitions(chromTransitionPosition, numChromTransitions);
+        }
+        catch (IOException e)
+        {
+            throw new RuntimeException(e);
+        }
+
+        if (null != chromTransitions)
```

Comment:
I don't think chromTransitions can ever be null. If you agree, please annotate parseTransitions() as @NotNull and remove the if()